### ISSUE TITLE
tflite: "Ensure that weights and inputs have the same channel dimension." in transpose_conv.cc.  It seems that the input and output must have same channel dimension?

### ISSUE THREAD

**Issue by merryImage** [tensorflow/tensorflow#21211]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**:
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by merryImage** [comment]
error info: tensorflow/contrib/lite/kernels/transpose_conv.cc:82 SizeOfDimension(input, 3) != SizeOfDimension(weights, 3) (1280 != 64)Node number 63 (TRANSPOSE_CONV) failed to prepare

**Comment by andrehentz** [comment]
There has been some change earlier this month related to that. Are you experiencing this when building from source?

**Comment by aselle** [comment]
What version of TensorFlow are you using? try pip installing tf-nightly and see if that works as @andrehentz mentions.

**Comment by minizon** [comment]
I've encountred the same problem when I'm instantiating an Interpreter from a tflite file in android.

java.lang.NullPointerException: Internal error: Cannot allocate memory for the interpreter: tensorflow/contrib/lite/kernels/transpose_conv.cc:82 SizeOfDimension(input, 3) != SizeOfDimension(weights, 3) (512 != 256)Node 54 failed to prepare

Indeed, when building the model I've changed the channels from 512 to 256 by using Conv2DTranspose in keras. I've referred to [tf_ops_compatibility](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/tf_ops_compatibility.md), it seems CONV_2D_TRANSPOSE is fully supported?

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by minizon** [comment]
I tried the [tf-nightly version](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/tutorials/post_training_quant.ipynb),  and used the [python api](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/toco/g3doc/python_api.md), it seems this kind of method works and this error did not happen.

**Comment by wt-huang** [comment]
@merryImage Closing this now, feel free to open a new issue if any other error comes up.

**Comment by yokey1219** [comment]
I find out that it caused by different axis-order between v1.9.0 and v1.10.0+.In file "tensorflow/contrib/lite/kernels/transpose_conv.cc",line 81,it is `TF_LITE_ENSURE_EQ(context,SizeOfDimension(input, 3),SizeOfDimension(weights, 0));`,but in v1.10.0 and above,it is `TF_LITE_ENSURE_EQ(context, SizeOfDimension(input, 3),SizeOfDimension(weights, 3));`.
So,you cannot use the diffrent version when you convert the model from tensorflow to lite.
